packages/astro/src/types/public/integrations.ts

```diff
@@ -248,10 +245,41 @@ export interface AstroIntegration {
  */
 export type IntegrationRouteData = Omit<
 	RouteData,
-	'isIndex' | 'fallbackRoutes' | 'redirectRoute'
+	'isIndex' | 'fallbackRoutes' | 'redirectRoute' | 'origin'
 > & {
 	/**
 	 * {@link RouteData.redirectRoute}
 	 */
 	redirectRoute?: IntegrationRouteData;
 };
+
+export interface IntegrationResolvedRoute {
+	/**
+	 * The current **pattern** of the route. For example:
+	 * - `src/pages/index.astro` has a pattern of `/`
+	 * - `src/pages/blog/[...slug].astro` has a pattern of `/blog/[...slug]`
+	 * - `src/pages/site/[blog]/[...slug].astro` has a pattern of `/site/[blog]/[...slug]`
+	 */
+	pattern: RouteData['route'];
+
+	/**
+	 *  Source component URL
+	 */
+	entrypoint: RouteData['component'];
+
+	/**
+	 * Whether the route is prerendered or not
+	 */
+	prerendered: RouteData['prerender'];
+
+	/**
+	 * Dynamic and spread route params
+	 * ex. "/pages/[lang]/[...slug].astro" will output the params ['lang', '...slug']
+	 */
+	params: RouteData['params'];
```

Comment:
I'll miss the `generate` function from `IntegrationRouteData` to generate URLs for a given route. Integrations will have to construct their own URL generator function based on the `pattern` and `params` which seems somewhat unnecessary since that function already exists in Astro and just isn't exposed.

It is not a blocker since it can be done in user-land

Reply:
I've included the minimum properties, but it's the opportunity to rethink this api tbh! So feel free to suggest things and we'll talk about it at the standup

Reply:
I think it makes sense to keep `generate`

Reply:
The only field used on official integrations/adapters that can't be recomputed on the consumer is `type` to indicate whether the route is an endpoint, page, redirect, or fallback.

- `pathname` is now `pattern`, so it is just a rename
- `segments` can be computed from `pattern`
- `generate` can be created from `segments`

So I think at least `type` should be present.

Reply:
I wonder if `segments` and `generate` should be properties on the object, or functions exported by Astro where you can pass `pattern`/`segments`